tachyon/zk/plonk/circuit/assignment.h

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_CIRCUIT_ASSIGNMENT_H_
+#define TACHYON_ZK_PLONK_CIRCUIT_ASSIGNMENT_H_
+
+#include <stddef.h>
+
+#include <optional>
+#include <string>
+
+#include "tachyon/base/functional/callback.h"
+#include "tachyon/math/base/rational_field.h"
+#include "tachyon/zk/plonk/circuit/challenge.h"
+#include "tachyon/zk/plonk/circuit/column.h"
+#include "tachyon/zk/plonk/circuit/selector.h"
+#include "tachyon/zk/plonk/error.h"
+
+namespace tachyon::zk {
+
+template <typename F>
+class Assignment {
+ public:
+  using AnnotationCallback = base::OnceCallback<std::string()>;
+  using AssignCallback = base::OnceCallback<math::RationalField<F>()>;
+
+  // Creates a new region and enters into it.
+  //
+  // Panics if we are currently in a region (if |ExitRegion()| was not called).
+  //
+  // Not intended for downstream consumption; use |Layouter::AssignRegion()|
```

Comment:
Do you know where |Layouter::AssignRegion()| is coming from? I can't find it from both tachyon and halo2 implementation.